paasta_tools/flink_tools.py

```diff
@@ -166,6 +189,71 @@ def _dashboard_get(cr_name: str, cluster: str, path: str) -> str:
     return response.text
 
 
+def _filter_for_endpoint(json_response: Any, endpoint: str) -> Mapping[str, Any]:
+    """
+    Filter json response to include only a subset of fields.
+    """
+    if endpoint == "config":
+        return {
+            key: value for (key, value) in json_response.items() if key in CONFIG_KEYS
+        }
+    if endpoint == "overview":
+        return {
+            key: value for (key, value) in json_response.items() if key in OVERVIEW_KEYS
+        }
+    if endpoint == "jobs":
+        return json_response
+    if endpoint.startswith("jobs"):
+        return {
+            key: value
+            for (key, value) in json_response.items()
+            if key in JOB_DETAILS_KEYS
+        }
+    return json_response
+
+
+def _get_jm_rest_api_base_url(cr: Mapping[str, Any]) -> str:
+    metadata = cr["metadata"]
+    cluster = metadata["labels"][paasta_prefixed("cluster")]
+    base_url = get_flink_ingress_url_root(cluster)
+
+    # dashboard_url = http://flink-jobmanager-host:port/paasta-service-cr-name
+    dashboard_url = urlparse(metadata["annotations"]["flink.yelp.com/dashboard_url"])
+    service_cr_name = dashboard_url.path
+
+    return urljoin(base_url, service_cr_name)
+
+
+def curl_flink_endpoint(cr_id: Mapping[str, str], endpoint: str) -> Mapping[str, Any]:
+    try:
+        cr = get_cr(settings.kubernetes_client, cr_id)
+        if cr is None:
+            raise ValueError(f"failed to get CR for id: {cr_id}")
+        base_url = _get_jm_rest_api_base_url(cr)
+
+        # Closing 'base_url' with '/' to force urljoin to append 'endpoint' to the path.
+        # If not, urljoin replaces the 'base_url' path with 'endpoint'.
+        url = urljoin(base_url + "/", endpoint)
```

Comment:
awww, i was misremembering how this worked ;_; - i could have sworn that urljoin would always append, but i guess not :p

(but at least this will make sure to dedupe `/` if base_url is ever updated to have a trailing `/` which is still nice)

Reply:
> (but at least this will make sure to dedupe / if base_url is ever updated to have a trailing / which is still nice)

That's correct